File: Python/Max Consecutive Ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_counter = 0
        counter = 0
        for num in nums:
            if num == 0:
                if counter > max_counter:
                    max_counter = counter
                counter = 0
            else:
                counter += 1
            
        return max_counter if max_counter > counter else counter
                

File: cpp/Max Consecutive Ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max = 0;
        int counter = 0;
        for (int temp : nums) {
            if (temp) {
                counter++;
            } else {
                max = max > counter ? max : counter;
                counter = 0;
            }
        }
        max = max > counter ? max : counter;
        return max;
    }
};
